test/test_buffer.cpp

```diff
@@ -188,14 +188,16 @@ struct holder
 	holder(holder const&) = delete;
 	holder& operator=(holder const&) = delete;
 	holder(holder&& rhs) noexcept : m_buf(rhs.m_buf), m_size(rhs.m_size) { rhs.m_buf = nullptr; }
-	holder& operator=(holder&& rhs) noexcept
+	// NOTE: move operator is commented to avoid warning due to
+	// -Wunused-member-function
+	/*holder& operator=(holder&& rhs) noexcept
 	{
 		if (m_buf) free_buffer(m_buf);
 		m_buf = rhs.m_buf;
 		m_size = rhs.m_size;
 		rhs.m_buf = nullptr;
 		return *this;
-	}
+	}*/
```

Comment:
hm. this doesn't look right. all the calls to ``append_buffer()`` below, that construct a temporary ``holder``, are  they must be moved in, right?

Reply:
oh, right. that's move constructed, not move assigned

Reply:
do you think it's a good idea to put an static assert with `std::is_move_constructible` for `Holder` in `chained_buffer` methods?

Reply:
well, I don't think it makes sense to test the test class just for the sake of it. Maybe ``chained_buffer`` would never invoke the move assignment of holders (in which case this should probably be ``= delete``). Or maybe there should be a test that attempts to move assign a ``chained_buffer`` to ensure it moves the holder correctly. Either way, those things seems like out of the scope of this patch. I'm just worried about maybe forgetting following up on this by just commenting it out

Reply:
Well, I would like to put `= delete`, but I need to take a look in detail, since there is a `#if TORRENT_CPP98_DEQUE` in `chained_buffer`, I want to know if that is still necessary.

Reply:
do you want me to move out this from the PR?

Reply:
if it works with ``= delete``, I would prefer that